packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -885,6 +891,18 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
     const newValue = newProps.value;
     workInProgress.memoizedProps = newProps;
 
+    const providerPropTypes = workInProgress.type.propTypes;
+
+    if (__DEV__ && providerPropTypes) {
```

Comment:
Could you please split this into two nested conditions? We try to always keep `if (__DEV__) {` as a separate block to make it very distinctive.